### PR TITLE
Backport

### PR DESCRIPTION
**Pull request by harenson**
Master: https://github.com/2600hz/kazoo/pull/6184

### CODE REVIEW

File: applications/crossbar/src/modules/cb_alerts.erl [hunk 1]
```diff
@@ -224,32 +224,82 @@ set_success_resp_status(Context) ->
...
-            lager:debug("unable to fetch port requests: ~p", [_R]),
-            Context;
-        {'ok', PortRequests} ->
-            check_port_requests(PortRequests, Context)
+-spec maybe_check_port_requests(cb_context:context(), kz_term:ne_binary()) -> cb_context:context().
+maybe_check_port_requests(Context, AccountId) ->
+    MasterId = cb_context:master_account_id(Context),
+    AuthId = cb_context:auth_account_id(Context),
+    case AuthId =:= MasterId of
+        'true' ->
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_alerts.erl]: Why not pattern-match here?

```erlang
case cb_context:auth_account_id(Context) of
    MasterAccountId -> ...;
    AuthId -> ...
end
```

File: applications/crossbar/test/cb_alerts_tests.erl [hunk 2]
```diff
@@ -28,45 +30,33 @@ cb_alerts_test_() ->
...
     Context = cb_context:set_resp_data(cb_context:new(), []),
-    {'ok', ActivePorts} = AccountActivePorts = account_active_ports(),
+    {'ok', ActivePorts} = account_active_and_unconfirmed_ports(),
 
     %% 1 submitted, 1 unconfirmed, and 1 rejected.
-    meck:expect(Mod, 'account_active_ports', fun(_) -> AccountActivePorts end),
-    Context1 = cb_alerts:check_port_requests(Context),
-    [Alert1, Alert2] = cb_context:resp_data(Context1),
+    Context1 = cb_alerts:do_check_port_requests(ActivePorts, Context),
+    Resp = cb_context:resp_data(Context1),
```
Comment (jamesaimonetti) [inline applications/crossbar/test/cb_alerts_tests.erl]: Same here, pattern-match: `[Alert1, Alert2] = Resp = cb_context:resp_data(Context1),` or get rid of Resp if unused.

File: core/kazoo_number_manager/src/knm_port_request.erl [hunk 3]
```diff
@@ -163,41 +165,57 @@ get_portin_number(AccountId, DID=?NE_BINARY) ->
...
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% @end
+%%------------------------------------------------------------------------------
+-spec account_ports_by_state(kz_term:ne_binary(), kz_term:ne_binary()) -> {'ok', kz_json:objects()} |
+          kz_datamgr:data_error().
+account_ports_by_state(AccountId, PortState) ->
+    ViewOptions = [{'startkey', [AccountId, PortState]}
+                  ,{endkey, [AccountId, PortState, kz_json:new()]}
```
Comment (jamesaimonetti) [inline core/kazoo_number_manager/src/knm_port_request.erl]: tickie

File: applications/crossbar/src/modules/cb_alerts.erl (hunk 1 above)
Comment (sergey-safarov) [reply applications/crossbar/src/modules/cb_alerts.erl]: and this string not need more
```erlang
AuthId = cb_context:auth_account_id(Context),
```

**Pull request merged**